lms/djangoapps/courseware/views/views.py

```diff
@@ -1635,6 +1635,12 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True, disable_sta
                 if not _check_sequence_exam_access(request, seq_block.location):
                     return HttpResponseForbidden("Access to exam content is restricted")
 
+        # AI Translations feature for course content, passes a translation language to blocks
+        # which can call through the ai_translation service (passed through XBlock runtime)
+        # to receive translated versions of course content.
+        if whole_course_translations_enabled_for_course(course_key):
+            student_view_context['translate_lang'] = request.GET.get('translate_lang')
```

Comment:
Do we really want to deal with blocks trying to access a missing service if this isn't enabled? Shouldn't this be like, something in the service itself that blocks can then check? Or something the service checks that no-ops if the feature is off or something?

Reply:
I had put this here to avoid sending that `translate_lang` which is currently what gates even calling the service in the first place, but it's probably better to always set that context and make the service / views themselves aware of the config setting.

Reply:
@jansenk , updated this to be on the actual XBlock "should I translate?" call.